### DISCUSSION TITLE
How to set the timezone?

### DISCUSSION THREAD

**Discussion by yuqiuwen** [mdomke/python-ulid#23] [Q&A]

**Comment by Avihais12344** [comment]
You can use the [`from_datetime`](https://python-ulid.readthedocs.io/en/latest/api.html#ulid.ULID.from_datetime) method, and give the datetime a timezone.

But if you want the format to include timezones, then I think you can't because of the [spec](https://github.com/ulid/spec).

**Reply by mdomke** [reply]
This is correct. The timestamp part of the ULID is in UNIX epoch, which is the number of seconds that have elapsed since January 1, 1970 (midnight UTC). When using the [from_datetime](https://python-ulid.readthedocs.io/en/latest/api.html#ulid.ULID.from_datetime)-classmethod the given datetime is automatically converted to UTC
```pycon
>>> dt = datetime(2024, 4, 2, 10, 20, 30, tzinfo=UTC)
>>> ULID.from_datetime(dt).datetime
datetime.datetime(2024, 4, 2, 10, 20, 30, tzinfo=datetime.timezone.utc)
``` 
In my case the `astimezon()`-method will convert the datetime to CEST, but as you can see, the timestamp is stil in UTC
```pycon
>>> dt.astimezone()
datetime.datetime(2024, 4, 2, 12, 20, 30, tzinfo=datetime.timezone(datetime.timedelta(seconds=7200), 'CEST'))
>>> ULID.from_datetime(dt.astimezone()).datetime
datetime.datetime(2024, 4, 2, 10, 20, 30, tzinfo=datetime.timezone.utc)
```
